### PR TITLE
Feature/prettier

### PR DESCRIPTION
**Pull request by patric-eberle**
## Pull request
This PR adds Prettier for auto code formatting. Please use this PR to check the implementation and test it by doing some additional commits.

e.g. by
- replacing `interface X {}` with `type X = {}` in some files
- fixing ESLint issues that have been temporary disabled in `.eslintrc.js`, to trigger the lint staged formatting

Please also check my walk-through on how to setup Prettier in an existing project: https://movento.atlassian.net/wiki/spaces/DUF/pages/4045308169/Prettier

Please note, that you might get ESLint issues based on the auto formatted code during commits. This issues might not appear in the file itself, because auto formatting is only applied if all the pre-commit lints succeed. See the known issues section in the walk-trough: https://movento.atlassian.net/wiki/spaces/DUF/pages/4045308169/Prettier#Unrelated-ESLint-errors

⚠️ Apply `npm run prettier` to the whole project before doing the final review and merge!
 
### Ticket
No ticket
 
### Browser testing
- http://localhost:5173/
 
### Checklist
- [x] I merged the current development branch (before testing)
- [x] Added JSDoc and styleguide demo
- [ ] Tested all links in project relevant browsers
- [ ] Tested all links in different screen sizes
- [x] Did run automated tests and linters
- [ ] Did assign ticket
- [x] Double checked target branch
 
## Review/Test checklist
- [ ] Did review code and documentation
- [ ] Tested all links in project relevant browsers
- [ ] Tested all links in different screen sizes
- [ ] Did check accessibility (Wave, only errors)
- [ ] Re-assign ticket to developer

### CODE REVIEW

File: .lintstagedrc.js [hunk 1]
```diff
@@ -1,9 +1,7 @@
+// NOTE: 'concurrent' had to be disabled to allow prettier to apply code changes before linting.
 module.exports = {
-  '*.{js,ts,vue}': [
-    () => 'vue-tsc --noEmit',
-    'eslint --config .eslintrc.fix.js --fix',
-  ],
-  '*.{css,vue,scss}': [
-    'stylelint --config .stylelintrc.fix.js --fix',
```
Comment (patric-eberle) [inline .lintstagedrc.js:7]: Todo: we should keep `stylelint --fix` because it applies property sorting for CSS, which is not supported by Prettier...

**Comment by benjamin-joham** [comment]
I would suggest adding the following rules to eslint too. In my opinion ESLint does a better(nicer) job on sorting imports than prettier does.
```javascript
    'sort-imports': ['error', { ignoreCase: true, ignoreDeclarationSort: true }],
    'import/order': [
      'error',
      {
        pathGroups: [
          {
            pattern: '@/**',
            group: 'internal',
          },
          {
            pattern: 'vue',
            group: 'builtin',
            position: 'before',
          },
        ],
        groups: [['builtin', 'external'], 'internal', ['parent', 'sibling', 'index'], 'type'],
        'newlines-between': 'always-and-inside-groups',
      },
    ],
    '@typescript-eslint/type-annotation-spacing': ['error'],
    '@typescript-eslint/space-before-blocks': ['error'],
    '@typescript-eslint/member-ordering': [
      'error',
      {
        default: [
          // Index signature
          'signature',

          // Fields
          'field',

          'private-static-method',
          'protected-static-method',
          'public-static-method',

          // Constructors
          'constructor',

          // Getters and Setters at the same rank
          ['get', 'set'],

          // Methods
          'private-method',
          'protected-method',
          'public-method',
        ],
      },
    ],
    'vue/max-attributes-per-line': [
      'error',
      {
        singleline: {
          max: 2, // should break with 3 or more
        },
        multiline: {
          max: 1,
        },
      },
    ],
``` 

Maybe we should also add `import eslintPlugin from 'vite-plugin-eslint';` to report ESLint issues to the browser.

**Comment by patric-eberle** [comment]
An additional issue I discovered: Changing only the code structure, when it will be reverted by prettier will run into a commit abort too, without any code being "reverted".

**Comment by patric-eberle** [comment]
Personal thoughts against Prettier:
- Tries to put as much on 1 line as possible. This does often make readability worse.
- Adds line breaks after equal sign (e.g. for `const foo = /n 'some long string....`)
- Places IF operators (`||`, `&&`) at the end of the line instead of the beginning, which makes conditions harder to read in my opinion.
- Enforces closing tag (`/>`) for Vue template elements, even if they don't need one (e.g. `<input />`, `<br />`, `` instead of `<input>`, `<br>`, ``).

**Comment by mob8607** [comment]
Looks very promising to me 👍 

Personally I agree with your concerns, but I feel like the advantage of prettier is bigger than the downside of these issues.

**Review by claudio147 — APPROVED** [review]

**Pull request merged**